nemo/collections/nlp/data/language_modeling/megatron/gpt_sft_dataset.py

```diff
@@ -40,6 +40,7 @@ def __init__(
         seed: int = 1234,
         context_key: str = "text",
         label_key: str = "answer",
+        query_key: Optional[str] = None,
```

Comment:
could you add query_key to the doc string below

Reply:
Sure! `query_key` is needed if user want to separate question from the context and construct their customized prompt.

Reply:
I think its best to leave the context field as is. We can merge the "query" into the context in a preprocessing step.

Reply:
When we have a extreme long context and we need to truncate them, we may truncate the query part. How can we put the query in the end and we still truncate the context part? I think most dataset can have the format with context, query, label.

Reply:
Dataset part is moved to this PR https://github.com/NVIDIA/NeMo/pull/7147